### PR TITLE
GEODE-2513 Unbrand .NET API docs section

### PR DESCRIPTION
**Pull request by karensmolermiller**
- Remove references to GemFire
- Namespace updates
- Correct typos and capitalization
- Use .NET scoping operator (not C++)
- Avoid duplication in the manual of QuickStart code

@davebarnes97 @joeymcallister @PivotalSarge @echobravopapa @dgkimura @mmartell 
Please give this PR a careful review.  The namespace change for .NET clients was tricky, and I'm not sure if the 'Generic' exists for Geode.  There were also a few places in the docs that used the C++ scoping operator (::) instead of a period; I think that those were typos in the original documentation.

### CODE REVIEW

File: docs/geode-native-docs/dotnet-caching-api/implementing-igfserializable.html.md.erb [hunk 1]
```diff
@@ -31,21 +31,21 @@ Examples follow the procedure.
...
 
-    The `DataInput` class represents the input stream and provides methods for reading input elements. The `FromData` function must read the elements of the input stream in the same order that they were written by `ToData`. For more about this, see the API documentation for `DataInput` at [http://gemfire-apis.docs.pivotal.io](http://gemfire-apis.docs.pivotal.io).
+    The `DataInput` class represents the input stream and provides methods for reading input elements. The `FromData` function must read the elements of the input stream in the same order that they were written by `ToData`.
 
 3.  Implement the `ClassId` function to return an integer which is unique for your class (in the set of all of your user-defined classes).
 
 ## Simple BankAccount Class
 
-This example shows a simple class, `BankAccount`, that encapsulates two `ints`: `customerId` and `accountId`:
+This example shows a simple class, `BankAccount`, that encapsulates the two `ints` `customerId` and `accountId`:
```
Comment (PivotalSarge) [inline docs/geode-native-docs/dotnet-caching-api/implementing-igfserializable.html.md.erb]: Maybe a comma between `ints` and `customerId`?

File: docs/geode-native-book/master_middleman/source/subnavs/geode-nc-nav.erb [hunk 2]
```diff
@@ -473,7 +473,7 @@ limitations under the License.
                                         <a href="/docs/guide-native/11/dotnet-caching-api/resolving-the-error.html">Resolving the Error</a>
                                     </li>
                                     <li>
-                                        <a href="/docs/guide-native/11/dotnet-caching-api/private-assembly.html">Using GemStone.GemFire.Cache.dll As a Private Assembly</a>
+                                        <a href="/docs/guide-native/11/dotnet-caching-api/private-assembly.html">Using Apache.Geode.Client.dll As a Private Assembly</a>
```
Comment (PivotalSarge) [inline docs/geode-native-book/master_middleman/source/subnavs/geode-nc-nav.erb]: I don't see `Apache.Geode.Client.dll`, only `Apache.Geode.dll`, after a Windows installation.

File: docs/geode-native-book/master_middleman/source/subnavs/geode-nc-nav.erb [hunk 3]
```diff
@@ -439,7 +439,7 @@ limitations under the License.
                                         </ul>
                                     </li>
                                     <li class="has_submenu">
-                                        <a href="/docs/guide-native/11/dotnet-caching-api/serialize-using-igfserializable.html">Serialize with the GemFire IGFSerializable Interface</a>
+                                        <a href="/docs/guide-native/11/dotnet-caching-api/serialize-using-igfserializable.html">Serialize with the Geode IGFSerializable Interface</a>
```
Comment (PivotalSarge) [inline docs/geode-native-book/master_middleman/source/subnavs/geode-nc-nav.erb]: I believe `IGFSerializable` has been replaced with `IGeodeSerializable`.

**Comment by karensmolermiller** [comment]
All corrections have been made.  I will merge this PR into develop as soon as the CI build finishes in the green.  Thanks for the helpful reviews.

**Comment by karensmolermiller** [comment]
Merged in to develop.

**Pull request closed without merging**